packages/scripts/src/admin-proposals/funding/0_Propose.ts

```diff
@@ -0,0 +1,141 @@
+// This script can be run against a mainnet fork by spinning a node in a separate terminal with:
+// HARDHAT_CHAIN_ID=1 yarn hardhat node --fork https://mainnet.infura.io/v3/<YOUR-INFURA-KEY> --port 9545 --no-deploy
+// and then running this script with:
+// GCKMS_WALLET=<OPTIONAL-GCKMS-WALLET> \
+// RECIPIENT=<FUNDING-RECEIVER-ADDRESS> \
+// TOKEN=<OPTIONAL-TOKEN-ADDRESS> \
+// AMOUNT=<FUNDING-AMOUNT> \
+// PROPOSAL_URL=<PROPOSAL-URL> \
+// yarn hardhat run ./src/admin-proposals/funding/0_Propose.ts --network <network>
+
+import { strict as assert } from "assert";
+import { BigNumber, Signer } from "ethers";
+import hre from "hardhat";
+import { getGckmsSigner } from "@uma/common";
+import { getAddress, ERC20Ethers, ProposerV2Ethers, VotingTokenEthers } from "@uma/contracts-node";
+import { getContractInstance } from "../../utils/contracts";
+import { AdminProposalTransaction } from "../../upgrade-tests/voting2/migrationUtils";
+
+const { ethers } = hre;
+
+require("dotenv").config();
+
+const proposerAddress = "0x2bAaA41d155ad8a4126184950B31F50A1513cE25";
+
+interface ProposalParams {
+  proposerSigner: Signer;
+  recipient: string;
+  tokenAddress: string;
+  amount: number;
+  proposalUrl: string;
+  trace: boolean;
+}
+
+async function getProposerSigner(env: NodeJS.ProcessEnv): Promise<Signer> {
+  let proposerSigner: Signer;
+
+  if (env.GCKMS_WALLET) {
+    proposerSigner = (await getGckmsSigner()).connect(ethers.provider);
+    if (proposerAddress.toLowerCase() != (await proposerSigner.getAddress()).toLowerCase())
+      throw new Error("GCKMS wallet does not match proposer wallet");
+  } else {
+    if (hre.network.name !== "localhost") throw new Error("Cannot impersonate on mainnet");
+    proposerSigner = await ethers.getImpersonatedSigner(proposerAddress);
+  }
+
+  return proposerSigner;
+}
```

Comment:
Do we have a shared common file for these scripts where this function could be implemented and shared?

Reply:
Not yet, but agree that we should do this for all admin proposals.
I'll create a separate issue for this

Reply:
Perfect!